### PR TITLE
Completing GET request for /api/articles/:article_id

### PR DESCRIPTION
**Pull request by charlie-mw**
- Created an endpoint for /api/articles/:article_id which responds with an article object 
- Created articles.models.js which includes a custom error for a non-existent article

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -56,3 +56,24 @@ describe("GET /api", () => {
...
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
+  });
```
Comment (nc-nick-jones) [inline __tests__/app.test.js:78]: we need some tests for correct error response (404 - valid but no existent article, 400 bad article_id format)

Comment (nc-rayhaan) [inline __tests__/app.test.js:78]: This is a lovely test! You are missing your error handling tests though! Please do add these in

File: models/articles.models.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+exports.selectArticleById = (article_id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .then((data) => {
+      if (data.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "Not found",
+        });
```
Comment (nc-nick-jones) [inline models/articles.models.js:11]: you have the behaviour! Just need a test for this.

**Review by nc-rayhaan — COMMENTED** [review]
Nice work! Please do add the missing tests when you get a chance

File: __tests__/app.test.js (hunk 1 above)
Comment (nc-nick-jones) [reply __tests__/app.test.js:78]: you may need to update your error handling middleware to pass these tests.

**Review by nc-nick-jones — COMMENTED** [review]
Nice work Charlie.  Could do with adding some more tests for cases you want to throw errors.  Please look at the notes, your rare treasures or can you handle this sprint for details.

**Pull request merged**